ksqldb-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -154,6 +155,7 @@ PullQueryResult executePullQuery(
       ConfiguredStatement<Query> statement,
       RoutingFilterFactory routingFilterFactory,
       RoutingOptions routingOptions,
+      ExecutorService executorService,
```

Comment:
Can we avoid passing down a general-purpose `ExecutorService`? It seems very easy to "abuse" if down the line someone needs a thread pool they'll think "oh look, I can just use this one that's already here!" and "steal" threads from the pull queries.

I'm about to hop on a meeting but I'll take a deeper look in a bit. I think it might be worth refactoring `HARouting` so that we can pass one instance down (or create on when we create the engine).

Reply:
Yes, I agree and I thought about it as well. The current problem is that the HARouting takes as argument the pull physical plan so that it can execute it when the query is served locally. I haven't come up with an idea on how to refactor this.

Reply:
You can pass the `PullPhysicalPlan` as an argument of `HARouting.handlePullQuery` rather than injecting into the constructor.  That should allow HARouting to be a singleton.